pkg/gatherers/conditional/validation.go

```diff
@@ -0,0 +1,147 @@
+package conditional
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/go-playground/validator"
+)
+
+var validate *validator.Validate
+
+func init() { //nolint:gochecknoinits
+	validate = validator.New()
+
+	validate.RegisterStructValidation(gatheringRuleValidator, GatheringRule{})
+	validate.RegisterAlias("openshift_namespace", "min=1,max=128,hostname,startswith=openshift-")
+}
+
+func validateGatheringRules(gatheringRules []GatheringRule) []error {
+	var allErrors []error
+
+	for _, gatheringRule := range gatheringRules {
+		errs := validateGatheringFunctionsParamsAreStructs(gatheringRule.GatheringFunctions)
+		if len(errs) > 0 {
+			allErrors = append(allErrors, errs...)
+			continue
+		}
+
+		err := validate.Struct(gatheringRule)
+		if err != nil {
+			errs = makeValidationErrorPretty(err)
+			allErrors = append(allErrors, errs...)
+			continue
+		}
+	}
+
+	return allErrors
+}
+
+func validateGatheringFunctionsParamsAreStructs(gatheringFunctions GatheringFunctions) (errs []error) {
+	// a workaround to not panic https://github.com/go-playground/validator/issues/789
+	for key, val := range gatheringFunctions {
+		if reflect.TypeOf(val).Kind() != reflect.Struct {
+			errs = append(errs, fmt.Errorf("gathering function params should be a struct, key is %v", key))
+		}
+	}
+	return
+}
+
+func gatheringRuleValidator(sl validator.StructLevel) {
+	gatheringRule := sl.Current().Interface().(GatheringRule)
+
+	for i := range gatheringRule.Conditions {
+		validateCondition(&gatheringRule.Conditions[i], sl)
+	}
+
+	if len(gatheringRule.GatheringFunctions) == 0 {
+		sl.ReportError(gatheringRule.GatheringFunctions, "GatheringFunctions", "GatheringFunctions", "not_empty", "")
+	}
+
+	for name, params := range gatheringRule.GatheringFunctions {
+		validateGatheringFunction(name, params, sl)
+	}
+}
+
+// validateCondition validates that a value of type ConditionWithParams is valid.
+// Valid value should have a valid Type and Params of the corresponding type
+func validateCondition(condition *ConditionWithParams, sl validator.StructLevel) {
+	if err := condition.Type.IsValid(); err != nil {
```

Comment:
I would leave this validation up to the JSON schema as well. We can define a set/enum of valid condition types.